x-pack/plugins/endpoint/common/types.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export class EndpointAppConstants {
+  static ENDPOINT_INDEX_NAME = 'endpoint-agent*';
+}
+
+export interface EndpointResultList {
+  // the endpoint restricted by the page size
+  endpoints: EndpointData[];
+  // the total number of unique endpoints in the index
+  total: number;
+  // the page size requested
+  request_page_size: number;
+  // the index requested
+  request_page_index: number;
+}
+
+export interface EndpointData {
+  machine_id: string;
+  created_at: Date;
```

Comment:
Q. I think this is a `string`, not a Date object ?? 🤷‍♂ _(same question for other date type of properties below)_

Reply:
Why do you think it is a str, is it because of the format.

Reply:
JSON does not have a way to serialize/de-serialize dates. They will come across as Strings if the value of attribute is an actual JS Date object (defaults to `(new Date()).toString()` output which is normally the localized date+time).
My guess is that you are getting a string from Elasticsearch's query response.

Reply:
@paul-tavares 
```
"event": {
        "created": "2020-01-24T21:01:56.183Z"
      },
```

Reply:
Thanks @nnamdifrankie . So back to the type - what will it be?

Reply:
@paul-tavares I think we can map it at time of use or parameter definition. Once the other PR is merged and the type definition is available. Remember that we use it here as output, so we can map it on output or the caller can. Also recall that this is an api route and can be called by any client curl, front end e.t.c so it is not critical here for us and makes no difference.

Reply:
Sorry @nnamdifrankie , I'm confused. Are you leaving the `created_at` defined as a `Date`? or `String`?

Reply:
I decided to define it a date so that it is explicit that when data is deserialized into that EndpointMetadata object the value of that field should be a date type. Is this satisfactory?

Reply:
That's fine. 
I'll follow up on the FE side as to how we want to handle this (ex. should we go through the known date fields we get from the API and convert them to Date object or just do the conversion when needed) (cc/ @oatkiller , @kevinlog )